pkg/commands/compute/setup/domain.go

```diff
@@ -157,3 +150,61 @@ func (d *Domains) validateDomain(input string) error {
 	}
 	return nil
 }
+
+func (d *Domains) createDomain(name string, attempt int) error {
+	if attempt > 1 {
+		d.Progress = text.ResetProgress(d.Stdout, d.Verbose)
+	}
+	d.Progress.Step(fmt.Sprintf("Creating domain '%s'...", name))
+
+	_, err := d.APIClient.CreateDomain(&fastly.CreateDomainInput{
+		ServiceID:      d.ServiceID,
+		ServiceVersion: d.ServiceVersion,
+		Name:           &name,
+	})
+	if err != nil {
+		if attempt > d.RetryLimit {
+			return fmt.Errorf("too many attempts")
+		}
+
+		// We have to stop the ticker so we can now prompt the user.
+		d.Progress.Fail()
+
+		if e, ok := err.(*fastly.HTTPError); ok {
+			if e.StatusCode == http.StatusBadRequest {
+				for _, he := range e.Errors {
+					// NOTE: In case the domain is already used by another customer.
+					// We'll give the user one additional chance to correct the domain.
+					if strings.Contains(he.Detail, "by another customer") {
+						var domain string
+						defaultDomain := generateDomainName()
+						if !d.AcceptDefaults && !d.NonInteractive {
+							text.Break(d.Stdout)
+							domain, err = text.Input(d.Stdout, text.BoldYellow(fmt.Sprintf("Domain already taken, please choose another (attempt %d or %d): [%s] ", attempt, d.RetryLimit, defaultDomain)), d.Stdin, d.validateDomain)
```

Comment:
is this meant to say "attempt x of y"? If so, the below suggestion fixes the typo -- 
```suggestion
							domain, err = text.Input(d.Stdout, text.BoldYellow(fmt.Sprintf("Domain already taken, please choose another (attempt %d of %d): [%s] ", attempt, d.RetryLimit, defaultDomain)), d.Stdin, d.validateDomain)
```